### ISSUE TITLE
Add clock for synchronous circuits

### ISSUE THREAD

**Issue by sparshg** [Spydr06/logicrs#38]
We need to implement the clock that can be operated on the chosen frequency. This will be useful in many other modules.

**Comment by Spydr06** [comment]
Yes, this seems really useful. If you want I can assign this issue to you

**Comment by sparshg** [comment]
I would have fixed it already instead of creating this issue 😅. I would need to understand more of your codebase to implement this I think.

**Comment by Spydr06** [comment]
Well, I won't pressure you xD
If you want to dig deeper and implement it, go ahead and ask me if you have any questions :)

**Comment by sparshg** [comment]
Also, it would be nice if you could add some things to contributing.md about how your project simulates the components, like how the updating of circuit is handled or maybe some other things that should be known to change the project's core.
